src/main/java/org/ow2/proactive/addons/cloud_watch/PCWUtility.java

```diff
@@ -0,0 +1,109 @@
+ * or a different license than the AGPL.
+ *
+ *  Initial developer(s):               The ProActive Team
+ *                        http://proactive.inria.fr/team_members.htm
+ *  Contributor(s):
+ *
+ *  * $$ACTIVEEON_INITIAL_DEV$$
+ */
+package org.ow2.proactive.addons.cloud_watch;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.ow2.proactive.addons.cloud_watch.model.RestResponse;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.client.fluent.Response;
+import org.apache.http.entity.ContentType;
+import org.apache.http.util.EntityUtils;
+
+public class PCWUtility {
+    private static final String PCW_REST_URL = "http://localhost:8080/proactive-cloud-watch/rules/";
+//    private static final String PCW_REST_URL = "http://localhost:8080/rules/";
+
+    /**
+     * method make Post call to REST API of Proactive Cloud Watch service for adding rule and starting polling
+     * @param  jsonRule a rule specified directly in json format
+     * @throws IOException
+     */
+    public static RestResponse addRule(String jsonRule) throws IOException {
+        Request request = Request.Post(PCW_REST_URL);
+        request.addHeader("Content-Type", "application/json");
+        request.bodyString(jsonRule, ContentType.TEXT_PLAIN);
+        return executeRequestAndCheck(request, HttpURLConnection.HTTP_CREATED);
+    }
+
+    /**
+     * method make Post call to REST API of Proactive Cloud Watch service for adding rule and starting polling
+     * @param  ruleFilePath local file path to rule
+     * @throws IOException
+     */
+    public static RestResponse addRuleFromFile(String ruleFilePath) throws IOException {
+        Request request = Request.Post(PCW_REST_URL);
+        request.addHeader("Content-Type", "application/json");
+        String contentRule = new String(Files.readAllBytes(Paths.get(ruleFilePath)));
+        request.bodyString(contentRule, ContentType.TEXT_PLAIN);
+        return executeRequestAndCheck(request, HttpURLConnection.HTTP_CREATED);
+    }
+
+    /**
+     * method make Delete call to REST API of Proactive Cloud Watch service for deleting rule and stop polling
+     * @param  ruleName a rule identifier
+     * @throws IOException
+     */
+    public static RestResponse deleteRule(String ruleName) throws IOException {
+        Request request = Request.Delete(PCW_REST_URL + "/" + ruleName);
+        request.addHeader("Content-Type", "application/json");
+        return executeRequestAndCheck(request, HttpURLConnection.HTTP_OK);
+    }
+
+    private static RestResponse executeRequestAndCheck(Request request, int httpOkStatus) throws IOException {
+        RestResponse response = executeRequest(request);
+        return checkResponseIsOK(response, httpOkStatus);
+    }
+
+    private static RestResponse checkResponseIsOK(RestResponse response, int successfulStatus) {
```

Comment:
Same here I think that the parameter successfulStatus is well named but I think that the method should be named checkResponseIsSucessful